ui/src/institutions/utils.js

```diff
@@ -0,0 +1,43 @@
+import { List } from 'immutable';
+import { makeCompliantMetaDescription } from '../common/utils';
+
+export function getPapersQueryString(recordId) {
+  return `authors.affiliations.record.$ref:${recordId}`;
+}
+
+function getAddressText(address) {
+  const placeName = address.get('place_name');
+  const city = address.getIn(['cities', 0]);
+  const state = address.get('state');
+  const country = address.get('country');
+  const addressElements = [placeName, city, state, country];
+  const addressText = addressElements.filter(Boolean).join(', ');
+  return addressText;
+}
+
+export function getInstitutionMetaDescription(institution) {
+  const ITEM_SEPARATOR = ' and ';
+
+  const legacyICNText = institution.get('legacy_ICN');
+  const institutionhierarchiesText = institution
+    .get('institution_hierarchy', List())
+    .map(
+      hierarchy =>
+        hierarchy.get('acronym')
+          ? `${hierarchy.get('name')} (${hierarchy.get('acronym')})`
+          : `${hierarchy.get('name')}`
+    )
+    .filter(Boolean)
```

Comment:
`name` can be missing too? if so, you should move `filter` above `map` like `filter(() => hierarchy.has('name'))`. this won't filter anything since it will get `"undefined"` not `undefined`